wagtail/admin/views/home.py

```diff
@@ -59,6 +59,32 @@ def render_html(self, parent_context: Mapping[str, Any] = None) -> str:
             return ""
 
 
+class WhatsNewInWagtailVersionPanel(Component):
+    name = "whats_new_in_wagtail_version"
+    template_name = "wagtailadmin/home/whats_new_in_wagtail_version.html"
+    order = 110
+    _version = "4"
+
+    def get_whats_new_banner_setting(self) -> Union[bool, str]:
+        return getattr(settings, "WAGTAIL_ENABLE_WHATS_NEW_BANNER", True)
+
+    def get_dismissible_id(self) -> str:
+        return f"{self.name}_{self._version}"
+
+    def get_context_data(self, parent_context: Mapping[str, Any]) -> Mapping[str, Any]:
+        return {"dismissible_id": self.get_dismissible_id()}
+
+    def render_html(self, parent_context: Mapping[str, Any] = None) -> str:
+        profile = getattr(parent_context["request"].user, "wagtail_userprofile", None)
+        if (
+            not self.get_whats_new_banner_setting()
+            or not profile
```

Comment:
A missing profile object would indicate that they're a brand new user who has not yet dismissed anything or changed any preferences from the default. Isn't that a situation where you _do_ want to show the banner?

Reply:
Ah yes, good catch. In the dismissibles view's `patch()` method, I previously didn't create the `UserProfile` object if it doesn't exist, so that meant there's no way for such user to dismiss the banner. I made some adjustments to use `UserProfile.get_for_user()` so that the profile is created if it doesn't exist, but I forgot to revisit the decision.